presto-thrift-tpch-node/src/main/java/com/facebook/presto/thrift/node/ThriftServerTpch.java

```diff
@@ -0,0 +1,462 @@
+import com.facebook.presto.thrift.interfaces.client.ThriftSessionValue;
+import com.facebook.presto.thrift.interfaces.client.ThriftSplit;
+import com.facebook.presto.thrift.interfaces.client.ThriftSplitBatch;
+import com.facebook.presto.thrift.interfaces.client.ThriftSplitsOrRows;
+import com.facebook.presto.thrift.interfaces.client.ThriftTableLayout;
+import com.facebook.presto.thrift.interfaces.client.ThriftTableLayoutResult;
+import com.facebook.presto.thrift.interfaces.client.ThriftTableMetadata;
+import com.facebook.presto.thrift.interfaces.client.ThriftTupleDomain;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriter;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriters;
+import com.facebook.presto.thrift.node.states.IndexInfo;
+import com.facebook.presto.thrift.node.states.LayoutInfo;
+import com.facebook.presto.thrift.node.states.SplitInfo;
+import com.facebook.presto.tpch.TpchMetadata;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import io.airlift.tpch.TpchColumn;
+import io.airlift.tpch.TpchColumnType;
+import io.airlift.tpch.TpchEntity;
+import io.airlift.tpch.TpchTable;
+
+import javax.annotation.Nullable;
+import javax.annotation.PreDestroy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.tests.AbstractTestIndexedQueries.INDEX_SPEC;
+import static com.facebook.presto.thrift.interfaces.readers.ColumnReaders.convertToPage;
+import static com.facebook.presto.tpch.TpchMetadata.ROW_NUMBER_COLUMN_NAME;
+import static com.facebook.presto.tpch.TpchMetadata.getPrestoType;
+import static com.facebook.presto.tpch.TpchRecordSet.createTpchRecordSet;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.util.concurrent.MoreExecutors.listeningDecorator;
+import static io.airlift.concurrent.Threads.threadsNamed;
+import static java.lang.Math.min;
+import static java.lang.Math.toIntExact;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.stream.Collectors.toList;
+
+public class ThriftServerTpch
+        implements ThriftPrestoClient
+{
+    private static final int DEFAULT_NUMBER_OF_SPLITS = 3;
+    private static final String NUMBER_OF_SPLITS_PARAMETER = "splits";
+    private static final List<String> SCHEMAS = ImmutableList.of("tiny", "sf1");
+    private static final int MAX_WRITERS_INITIAL_CAPACITY = 8192;
+
+    private final ObjectMapper mapper = new ObjectMapper();
```

Comment:
Making this static would allow a couple of other functions to become static also.